src/OperationalInsights/OperationalInsights/Client/OperationalInsightsClient.Clusters.cs

```diff
@@ -94,21 +94,6 @@ public virtual PSCluster UpdatePSCluster(string resourceGroupName, string cluste
 
             parameters.Sku = parameters.Sku != null ? parameters.Sku : existingCluster.Sku;
 
-            if (parameters.KeyVaultProperties != null)
```

Comment:
please add this if block back

Reply:
this block is redundant. please look at PSKeyVaultProperties.CreateProperties() - the validation moved there

Reply:
can you point me where you ensure PSClusterPatch.KeyVaultProperties.KeyName has a value?

Reply:
please look at the calling function at UpdateAzureOperationalInsightsClusterCommand.cs -> ExecuteCmdlet()
the KeyVaultProperties are set by using PSKeyVaultProperties.CreateProperties(this.KeyVaultUri, this.KeyName, this.KeyVersion);
it ensures that if not params were passed a null value will be set.
when passing the value to the .net SDK, if we pass this value (keyVaultProperties) with any value rather then null - our backend will try and set these new null values and the request will return with bad request. so only if at least one of the values of keyVaultProperties is set - we will pass on the keyVaultProperties object